sockeye/transformer.py

```diff
@@ -291,14 +293,15 @@ def __init__(self, max_length: int) -> None:
     def forward(self, is_train, req, in_data, out_data, aux):
```

Comment:
transformer.py also contains `AutoRegressiveBias` which also uses a large negative value to ensure auto-regressiveness in the Transformer Decoder. This also needs to be fixed.

Reply:
Its the same argument as in my comment above. you need to guard against hitting the fp16 range at masking time, and when using the bias tensor in the model.

Reply:
added unit test and support for dtype argument